### ISSUE TITLE
False positive of `Credo.Check.Warning.UnusedStringOperation`

### ISSUE THREAD

**Issue by am-kantox** [rrrene/credo#1024]
### Precheck

* ✓ the bug has not yet been reported here

### Environment

* Credo version (`mix credo -v`): **1.6.7-ref.develop.c791057**
* Erlang/Elixir version (`elixir -v`): **Erlang/OTP 25 [erts-13.1.3] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit:ns], Elixir 1.14.3 (compiled with Erlang/OTP 25)**
* Operating system: **Linux** / **MacOS**

### What were you trying to do?

```
┃   [W] Category: warning 
┃    ↗  Priority: high 
┃ 
┃       There should be no unused return values for String functions.
┃       lib/foo/bar.ex:87:44 (Foo.Bar.baz)
┃ 
┃    __ CODE IN QUESTION
┃ 
┃    85             dynamically_get_module().f(
┃    86               "values",
┃    87               value |> Atom.to_string() |> String.upcase()
┃                                                  ^^^^^^^^^^^^^
┃    88             )
┃    89         end
┃       
┃    __ WHY IT MATTERS
┃ 
┃       The result of a call to the String module's functions has to be used.
```

### Expected outcome

No warning.

### Actual outcome

Warning on dangling `|> String.upcase()` _within_ the function call parentheses if the outer function is applied to a dynamically retrived module.

### Test to reproduce

```elixir
  test "it should NOT report a violation when inside of function argument" do
    """
    defmodule CredoSampleModule do
    def runner, do: __MODULE__
    def do_some_function(parameter), do: IO.puts(parameter)
    def some_function(parameter) do
      CredoSampleModule.runner().do_some_function(parameter |> String.upcase())
      :ok
    end
    end
    """
    |> to_source_file
    |> run_check(@described_check)
    |> refute_issues()
  end
```

### Comment

The warning goes away if refactor it as

```elixir
values = value |> Atom.to_string() |> String.upcase()
f("values", values)
```

**Comment by rrrene** [comment]
Thx for reporting! :+1:

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by am-kantox** [comment]
Works like a charm! 

Thank you.
